libs/sdk-ui-dashboard/src/topBar/DashboardTitle.tsx

```diff
@@ -11,19 +13,62 @@ export interface IDashboardTitleProps {
     title: string;
 
     /**
-     * Callback that the component MUST call when the title is clicked.
+     * Callback that the component calls when title edit enabled the title is changed. If not set, title
+     * taken as non-editable.
      */
-    onTitleClicked: () => void;
+    onTitleChanged?: (title: string) => void;
+
+    /**
+     * Indicates if title edit is enabled.
+     */
+    isEditEnabled?: boolean;
 }
 
 /**
  * @internal
  */
-export const DashboardTitle: React.FC<IDashboardTitleProps> = (_props: IDashboardTitleProps) => {
-    return null;
+export const DashboardTitleCore: React.FC<IDashboardTitleProps> = (props: IDashboardTitleProps) => {
+    const { title, isEditEnabled, onTitleChanged } = props;
+
+    const renderEditableTitle = () => {
+        // if onTitleChanged callback not set, dashboard is not editable.
+        if (!isEditEnabled || !onTitleChanged) {
```

Comment:
So if `isEditEnabled && !onTitleChanged` this component will render null -> no title is shown. That is most likely wrong.